[elliot-tang] iP

elliot-tang opened the pull request:
# Orion

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Orion frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Orion {
    public static void main(String[] args) {
        new Orion().run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

elliot-tang committed:
Rename main class

elliot-tang committed:
Add greeting

elliot-tang committed:
Update READ.ME

elliot-tang committed:
Add echo functionality

elliot-tang committed:
Add add and list functionalities

elliot-tang committed:
Add Task class

elliot-tang committed:
Add mark as done functionality

elliot-tang committed:
Add support for todo, deadline and event task types

elliot-tang committed:
Update UI test case

elliot-tang committed:
Add exception handling

elliot-tang committed:
Add delete task functionality

elliot-tang committed:
Add load task functionality

elliot-tang committed:
Add support for storing completed status of tasks

elliot-tang committed:
Add save tasks to file functionality

elliot-tang committed:
Add error handling for folder not found case

elliot-tang committed:
Merge branch 'branch-Level-7'
User task list does not save across logins.

Saving user task lists to a file after every
login is more user-friendly so that they do
not need to key in all their tasks again.

Let's
* add a feature to save task lists to a file upon logout
* add a feature to read task lists from files upon login

elliot-tang committed:
Add support for LocalDate

elliot-tang committed:
Fix bug with saving new date format

elliot-tang committed:
Merge branch 'branch-Level-8'
Dates for Deadlines and Events stored as Strings.

Storing dates as LocalDates enables input checking
to prevent invalid dates from being input, and
enables operations to be done on dates.

Let's update the fields in Deadline and Event classes
to use LocalDate instead of String.

elliot-tang committed:
Add packages

elliot-tang committed:
Remove duplicate files

elliot-tang committed:
Improve error handling

elliot-tang committed:
Refactor code for more OOP

elliot-tang committed:
Merge branch 'branch-A-MoreOOP'
Current functionality of Orion mostly stored in
a single file.

Extracting out closely related code into classes
and packages will help improve organisation and
maintainability of code.

Let's
1. extract out closely related code into classes
1. organise files into packages

elliot-tang committed:
Merge branch 'add-gradle-support'

elliot-tang committed:
Add test cases

elliot-tang committed:
Configure build.gradle for JAR file creation

elliot-tang committed:
Add JavaDocs

elliot-tang committed:
Add functionality to find task by keyword

elliot-tang committed:
Merge branch 'branch-A-JavaDoc'
Current code has no inline documentation.

This can make the code difficult to understand.

Let's add JavaDocs to generate documentation for future reference.

elliot-tang committed:
Modify TaskList file to adhere to coding principles

elliot-tang committed:
Merge branch 'branch-A-CodingStandard'
Current code does not follow coding standard in
certain parts.

This can make reading code more difficult.

Let's change the code to adhere to coding standards.

elliot-tang committed:
Merge branch 'branch-Level-9

marchjlim commented on src/main/java/orion/chatbot/Parser.java:
```diff
@@ -0,0 +1,132 @@
+    protected static Command parse(String input) throws OrionException {
+        String[] inputArray = input.split(" ");
+        String command = inputArray[0].toLowerCase();
+        String[] parsed = input.split("/");
```
Perhaps the naming could be changed to parsedInputArray here, for clarity?

marchjlim commented on src/main/java/orion/chatbot/Ui.java:
```diff
@@ -0,0 +1,197 @@
+     *
+     * @param taskNo The number of tasks loaded.
+     */
+    protected void printLoadTasks(int taskNo) {
```
Please make the method name a little less ambiguous, as printLoadTasks can be interpreted as a Task might be printed.

marchjlim commented on src/main/java/orion/commands/DeleteTaskCommand.java:
Clean code! Overall pretty easy to read and understand

marchjlim commented on src/main/java/orion/commands/AddTaskCommand.java:
Overall great class names for all Command classes! Easy to understand their functionalities as well.

marchjlim commented on src/main/java/orion/exceptions/OrionTaskListSaveException.java:
```diff
@@ -0,0 +1,20 @@
+     *
+     * @param message the detail message that describes the reason for this exception
+     */
+    public OrionTaskListSaveException(String message) {
```
Perhaps the name could be OrionSaveTaskListException?

marchjlim reviewed (commented):
LGTM. Just some nits to fix.

elliot-tang committed:
Add support for checkstyle and fixed style issues

elliot-tang committed:
Fix more style errors

elliot-tang committed:
Merge branch 'branch-A-CheckStyle'

audipras commented on src/main/java/orion/chatbot/Orion.java:
```diff
@@ -0,0 +1,88 @@
+        while (isOnline) {
+            try {
+                String command = ui.readCommand();
+                Command c = Parser.parse(command);
+                c.execute(tasks, storage, ui);
+                isOnline = !c.isExit();
```
perhaps a more intuitive name like command would be better here?

audipras commented on src/main/java/orion/tasks/Event.java:
perhaps the task class names should share a "Task" suffix?

audipras commented on src/main/java/orion/chatbot/Ui.java:
I like how the method names all share the print prefix!

audipras commented on src/main/java/orion/exceptions/OrionException.java:
```diff
@@ -0,0 +1,21 @@
+package orion.exceptions;
+
+/**
+ * Represents a base exception class for all exceptions related to the Orion application.
+ * <p>
+ * This abstract class extends {@link Exception} and is used as a base class for
+ * custom exceptions in the Orion application. It allows for the creation of specific
+ * exception types with custom error messages.
+ * </p>
+ */
```
I like the javadocs for the abstract OrionException class! Concise, descriptive, and tells you its purpose.

audipras reviewed (commented):
LGTM! Your code follows appropriate naming conventions, which is great for readability and maintainability. A few really minor nits to fix here and there, however excellent job overall!

elliot-tang committed:
Add GUI

elliot-tang committed:
Merge branch 'branch-Level-10'

elliot-tang committed:
Remove dead code

There are several lines of code which are not used
and are commented out.

This can cause the code to be confusing to read.

Let's remove all instances of dead code.

elliot-tang committed:
Reduce Parser method length

The parse method of Parser is very long.

This can make reading the code difficult and tedious.

Let's adhere to SLAP principles by abstracting out low level tasks
into subclasses, thereby reducing the length of the parse method.

elliot-tang committed:
Refactor Storage

The loadTask method in Storage is very long.

This can make it very hard to read and understand the code.

Let's refactor the method and extract out related code into
separate methods to adhere to the SLAP principle and improve
code quality.

elliot-tang committed:
Add assert statements

elliot-tang committed:
Merge pull request #2 from elliot-tang/branch-A-Assertions

Add assert statements

Current code is missing assert statements.

Assert statements can help greatly with defining and checking
assumptions made about the program state, thereby helping
detect programmer errors.

Let's add assert statements into the code to improve code safety.

elliot-tang committed:
Merge branch 'master' into branch-A-CodeQuality

elliot-tang committed:
Improve code quality

There are several very long methods and instances of dead code.

This can make the code difficult to read and understand.

Let's refactor the long methods and extract similar functionality
out into shorter methods, and delete all instances of dead code
to improve code quality.

elliot-tang committed:
Merge branch 'master' of https://github.com/elliot-tang/ip

elliot-tang committed:
Add help command

elliot-tang committed:
Add help command

elliot-tang committed:
Improve test coverage

elliot-tang committed:
Improve style and update Javadocs

elliot-tang committed:
Add user guide

elliot-tang committed:
Add image attributions
